app/src/app/components/stix/stix-list/stix-list.component.ts

```diff
@@ -672,16 +689,42 @@ export class StixListComponent implements OnInit, AfterViewInit, OnDestroy {
      * Apply all controls and fetch objects from the back-end if configured
      */
     public applyControls() {
+        const {deprecated, revoked, state, platforms, domains, exclusiveDeprecated, exclusiveRevoked} = this.getFilterObjectStates();
         if ("stixObjects" in this.config) {
             if (this.config.stixObjects instanceof Observable) {
                 // pull objects out of observable
             } else {
                 // filter on STIX objects specified in the config
                 let filtered = this.config.stixObjects;
 
-                // deprecation filter
-                if (this.config.showDeprecatedFilter && !this.filter.includes("state.deprecated")) {
-                    filtered = filtered.filter(o => !o.deprecated)
+                //filter by domains
+                if (Array.isArray(domains) && domains.length > 0) {
+                  filtered = filtered.filter((obj:any)=> obj.domains.some(domains));
+                }
+
+                //filter by platforms
+                if (Array.isArray(platforms) && platforms.length > 0) {
+                  filtered = filtered.filter((obj:any)=> obj.platforms.some(platforms));
+                }
```

Comment:
An error is thrown when filtering objects by domain or platform:

<img width="388" alt="image" src="https://github.com/center-for-threat-informed-defense/attack-workbench-frontend/assets/48832936/4b4a8396-c661-4510-8fbf-1e2860415802">